### ISSUE TITLE
sports analytics code

### ISSUE THREAD

**Issue by spartancoder1154** [spartancoder1154/chat-bot-with-python3#12]
import codecademylib3_seaborn
import matplotlib.pyplot as plt
from sklearn.svm import SVC
from sklearn.model_selection import train_test_split
from svm_visualization import draw_boundary
from players import aaron_judge, jose_altuve, david_ortiz

# Step 1: Inspect columns (optional)
print(aaron_judge.columns)

# Step 2: Inspect description values (optional)
print(aaron_judge.description.unique())

# Step 3: Inspect type values (optional)
print(aaron_judge.type.unique())

# Step 4: Map labels: S -> 1, B -> 0
aaron_judge['type'] = aaron_judge['type'].map({'S':1, 'B':0})

# Step 5: Check mapping
print(aaron_judge['type'])

# Step 6: Inspect plate_x (optional)
print(aaron_judge['plate_x'])

# Step 7: Drop NaNs in relevant columns
aaron_judge = aaron_judge.dropna(subset=['plate_x','plate_z','type'])

# Step 8: Scatter plot of pitches
fig, ax = plt.subplots()
plt.scatter(
    x=aaron_judge['plate_x'],
    y=aaron_judge['plate_z'],
    c=aaron_judge['type'],
    cmap=plt.cm.coolwarm,
    alpha=0.25
)

# Step 9: Split into training and validation sets
training_set, validation_set = train_test_split(aaron_judge, random_state=1)

# Step 10: Create SVM classifier
classifier = SVC(kernel='rbf')

# Step 11: Fit classifier
classifier.fit(
    training_set[['plate_x','plate_z']],
    training_set['type']
)

# Step 12: Draw decision boundary
draw_boundary(ax, classifier)

# Step 13: Evaluate accuracy
accuracy = classifier.score(
    validation_set[['plate_x','plate_z']],
    validation_set['type']
)
print("Validation Accuracy:", accuracy)

plt.show()
